compiler/src/dotty/tools/dotc/parsing/JavaParsers.scala

```diff
@@ -1080,24 +1080,30 @@ object JavaParsers {
     /** CompilationUnit ::= [package QualId semi] TopStatSeq
       */
     def compilationUnit(): Tree = {
-      val start = in.offset
+      val buf = ListBuffer.empty[Tree]
+      var start = in.offset
+      val leadingAnnots = if (in.token == AT) annotations() else Nil
       val pkg: RefTree =
-        if (in.token == AT || in.token == PACKAGE) {
-          annotations()
+        if in.token == PACKAGE then
+          if !leadingAnnots.isEmpty then
+            //if (unit.source.file.name != "package-info.java")
+            //  syntaxError(pos, "package annotations must be in file package-info.java")
```

Comment:
not sure scalac needs to enforce this javac rule if the annotations are not used

Reply:
I think generally we don't add code to enforce Java rules that don't affect us in any way.